CycloneDX/Services/SolutionFileService.cs

```diff
@@ -96,7 +97,17 @@ public async Task<HashSet<DotnetDependency>> GetSolutionDotnetDependencys(string
 
             var packages = new HashSet<DotnetDependency>();
 
-            var projectPaths = await GetSolutionProjectReferencesAsync(solutionFilePath).ConfigureAwait(false);
+            HashSet<(string path,bool isTestProject)> projectPaths = new();
+
+            AnalyzerManager manager = new AnalyzerManager(solutionFilePath);
+            foreach (var (key, project) in manager.Projects)
+            {
+                var buildResults = project.Build();
+                var isTestProject = buildResults.SelectMany(x => x.Properties)
+                    .Any(x => x.Key.Equals("IsTestProject", StringComparison.OrdinalIgnoreCase) && x.Value.Equals("true", StringComparison.OrdinalIgnoreCase));
+
```

Comment:
test projects should not be examined  if excludeTestProjects == true therefore should not be added